source/common/http/conn_manager_impl.cc

```diff
@@ -141,14 +145,14 @@ void ConnectionManagerImpl::initializeReadFilterCallbacks(Network::ReadFilterCal
   read_callbacks_->connection().addConnectionCallbacks(*this);
 
   if (config_.idleTimeout()) {
-    connection_idle_timer_ = read_callbacks_->connection().dispatcher().createTimer(
-        [this]() -> void { onIdleTimeout(); });
+    connection_idle_timer_ =
+        overload_timer_factory_(kConnectionIdleTimerLabel, [this]() -> void { onIdleTimeout(); });
```

Comment:
Should we consider some changes to how timers are created and enabled?

The thing to realize here is the pattern of creating a timer and immediately enabling it.  Should the factory create and enable the timer internally?  Doing that would provide us with some freedom regarding representation of timers that have similar deadline deltas.  The current Timer class that accepts a delta on enableTimer, we may want a class that takes the timeout delta as a constructor argument and provides methods to register/unregister for that timeout.

Reply:
I'm not against changing the timer interface, but I think that should be addressed in another PR. That would also allow us to get rid of the redundancies of the `enableTimer` and `enableHRTimer` methods - if you construct a timer with a high-resolution value, it is a HR timer, otherwise a regular timer.

Reply:
You can call enableTimer with different values on the same timer object.

The issue of standard vs HR timers is separate.  I think that based on how timers are usually created we only need enableHRTimer.  I have a pending change that may obsolete the need for enableHRTimer.